src/lm/arpa-file-parser.h

```diff
@@ -86,9 +91,35 @@ class ArpaFileParser {
   void Read(std::istream &is, bool binary);
 
   /// Parser options.
-  const ArpaParseOptions& Options() const { return options_; }
+  const ArpaParseOptions &Options() const { return options_; }
 
  protected:
+  /// A status value returned from ConsumeNGram() overload.
+  class Status {
+   public:
+    /// Return the warninig status to print a warninig and continue.
+    static Status Warning(const std::string& message) {
+      return Status(kWarninig, message); }
```

Comment:
The origin is an issue, I agree. Let me think what I can do to have both good diags and local reporting.